🐛: Cannot compile on FreeBSD - v0.12.5

### What happened?

Hello,
I could compile stalwart 0.12.4 on FreeBSD, but 0.12.5 throws an error:

   Compiling jemallocator v0.5.4
   Compiling smtp v0.12.5 (/root/stalwart-0.12.5/crates/smtp)
   Compiling imap v0.12.5 (/root/stalwart-0.12.5/crates/imap)
   Compiling pop3 v0.12.5 (/root/stalwart-0.12.5/crates/pop3)
   Compiling managesieve v0.12.5 (/root/stalwart-0.12.5/crates/managesieve)
warning: `email` (lib) generated 1 warning (run `cargo fix --lib -p email` to apply 1 suggestion)
   Compiling services v0.12.5 (/root/stalwart-0.12.5/crates/services)
   Compiling migration v0.12.5 (/root/stalwart-0.12.5/crates/migration)
warning: unused import: `smtp::reporting::SmtpReporting
  --> crates/services/src/housekeeper/mod.rs:28:5
   |
28 | use smtp::reporting::SmtpReporting;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `logo_resource` found for reference `&common::Server` in the current scope
   --> crates/services/src/task_manager/alarm.rs:129:29
    |
129 |     let logo = match server.logo_resource(account_main_domain).await {
    |                             ^^^^^^^^^^^^^ method not found in `&Server`

error[E0609]: no field `content_type` on type `&_`
   --> crates/services/src/task_manager/alarm.rs:141:35
    |
141 |             ContentType::new(logo.content_type.as_ref()),
    |                                   ^^^^^^^^^^^^ unknown field

error[E0599]: no method named `logo_resource` found for reference `&common::Server` in the current scope
   --> crates/services/src/task_manager/imip.rs:117:29
    |
117 |     let logo = match server.logo_resource(sender_domain).await {
    |                             ^^^^^^^^^^^^^ method not found in `&Server`

error[E0609]: no field `content_type` on type `&_`
   --> crates/services/src/task_manager/imip.rs:130:35
    |
130 |             ContentType::new(logo.content_type.as_ref()),
    |                                   ^^^^^^^^^^^^ unknown field

Some errors have detailed explanations: E0599, E0609.
For more information about an error, try `rustc --explain E0599`.
warning: `services` (lib) generated 1 warning
error: could not compile `services` (lib) due to 4 previous errors; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
warning: `common` (lib) generated 13 warnings (run `cargo fix --lib -p common` to apply 9 suggestions)


Thank you!

### How can we reproduce the problem?

I can reproduce the problem by doing the following steps:

### Version

v0.12.x

### What database are you using?

None

### What blob storage are you using?

None

### Where is your directory located?

None

### What operating system are you using?

FreeBSD

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -401,8 +401,8 @@ jobs:
         include:
           - target: x86_64-unknown-freebsd
             arch: x86_64
-          - target: aarch64-unknown-freebsd
-            arch: aarch64
+          # - target: aarch64-unknown-freebsd
+          #   arch: aarch64
     steps:
       - name: Checkout
         uses: actions/checkout@v6.0.2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -391,6 +391,59 @@ jobs:
           name: artifact-${{matrix.target}}
           path: artifacts
 
+  freebsd:
+    name: Build / ${{matrix.target}}
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: x86_64-unknown-freebsd
+            arch: x86_64
+          - target: aarch64-unknown-freebsd
+            arch: aarch64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v6.0.2
+
+      - name: Build in FreeBSD VM
+        uses: vmactions/freebsd-vm@v1
+        with:
+          release: "15.1"
+          arch: ${{matrix.arch}}
+          usesh: true
+          mem: 14336
+          cpu: 4
+          sync: rsync
+          copyback: true
+          # gmake: required by jemalloc-sys on BSD hosts
+          # llvm: provides libclang for bindgen (librocksdb-sys)
+          # rust: libsqlite3-sys 0.38 uses cfg_select!, stabilized in Rust
+          # 1.95. The default 'quarterly' pkg repo still ships rust 1.94, so
+          # switch to the 'latest' repo (currently 1.96.1). rustup is not an
+          # option here: aarch64-unknown-freebsd has no rustup toolchains yet.
+          prepare: |
+            set -e
+            mkdir -p /usr/local/etc/pkg/repos
+            echo 'FreeBSD: { url: "pkg+https://pkg.freebsd.org/${ABI}/latest", mirror_type: "srv" }' > /usr/local/etc/pkg/repos/FreeBSD.conf
+            pkg update -f
+            pkg install -y rust gmake llvm rocksdb
+            rustc --version
+          run: |
+            set -e
+            export CARGO_TARGET_DIR=/tmp/target
+            export CARGO_TERM_COLOR=always
+            cargo build --release -p stalwart --no-default-features --features "sqlite postgres mysql rocks s3 redis azure nats enterprise"
+            mkdir -p artifacts
+            cp /tmp/target/release/stalwart artifacts/stalwart
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v7
+        with:
+          name: artifact-${{matrix.target}}
+          path: artifacts
+
   release:
     name: Release
     permissions:
@@ -398,7 +451,7 @@ jobs:
       contents: write
       attestations: write
     if: github.event_name == 'push' || inputs.Release
-    needs: [linux, windows, macos]
+    needs: [linux, windows, macos, freebsd]
     runs-on: ubuntu-latest
     steps:
       # Must run before artifacts are downloaded — checkout cleans the workspace.
@@ -484,7 +537,7 @@ jobs:
 
   publish:
     name: Publish release
-    needs: [linux, windows, macos, multiarch, release]
+    needs: [linux, windows, macos, freebsd, multiarch, release]
     if: startsWith(github.ref, 'refs/tags/')
     runs-on: ubuntu-latest
     permissions:
@@ -497,7 +550,7 @@ jobs:
 
   cleanup:
     name: Cleanup failed release
-    needs: [linux, windows, macos, multiarch, release]
+    needs: [linux, windows, macos, freebsd, multiarch, release]
     if: failure() && startsWith(github.ref, 'refs/tags/')
     runs-on: ubuntu-latest
     permissions:

